python/ray/data/datasource/parquet_datasource.py

```diff
@@ -84,33 +87,35 @@ def read_pieces(serialized_pieces: List[str]) -> pa.Table:
 
             from pyarrow.dataset import _get_partition_keys
 
+            ctx = DatasetContext.get_current()
+            output_buffer = BlockOutputBuffer(
+                block_udf=_block_udf,
+                target_max_block_size=ctx.target_max_block_size)
+
             logger.debug(f"Reading {len(pieces)} parquet pieces")
             use_threads = reader_args.pop("use_threads", False)
-            tables = []
             for piece in pieces:
-                table = piece.to_table(
+                part = _get_partition_keys(piece.partition_expression)
+                batches = piece.to_batches(
                     use_threads=use_threads,
                     columns=columns,
                     schema=schema,
                     **reader_args)
```

Comment:
It looks like the default batch size for the scanner used here is 1 million rows; do you think that's a good default batch size? We've typically seen an in-memory bytes-per-row of 100-800 bytes (most in the 150-200 bytes range), which results in ~95-750 MiB for a 1 million row batch, so this batch size seems pretty reasonable to me.

https://github.com/apache/arrow/blob/30ac30fd0e375bae1e57b99e38b782e634773fbe/python/pyarrow/_dataset.pyx#L2975

Reply:
Hmm changed to 100k by default just in case.

Reply:
We should probably revisit this in the future in case of huge rows, maybe by calculating the row size via schema, but I think it's ok for now.